vercel/examples
# add an example with web3 and nextauth.js

### Description

Often applications using Metamask to authenticate users resort to fetchig everything from the client as Metamask works by injecting itself on the window object. This means these apps are missing a lot of what Next.js has to offer. The techniques in this example allow users to get their connected users wallet and use them with great Next.js features

- `middleware`
- `getServerSideProps`
- Eventually RSC

[Readme of the example](https://github.com/vercel/examples/tree/feat/web3-nextauthjs/solutions/web3-sessions)

### Best Way to Test

1. Get the metamask extension and setup a wallet
2. Try navigating to `/protected` you should be redirected to `/`
3. Login, authorize the app in metamask
4. On `/protected` try switching to a new sub wallet or forcing a metamask disconnect. You will be redirected to `/`
5. try the regular logout 

### Type of Change


### New Example Checklist
